LF-3911: create side menu expanded version

antsgar opened the pull request:
**Description**

- Implemented new desktop and mobile navigation.
- Refactored finances routes so that they're all nested in /finances to reflect new IA.

Jira link:
https://lite-farm.atlassian.net/browse/LF-3911

**Type of change**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Passes test case
- [x] UI components visually reviewed on desktop view
- [ ] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [ ] I have added "MISSING" for all new language tags to languages I don't speak
- [x] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

Duncan-Brain commented on packages/webapp/src/Routes.jsx:
```diff
@@ -316,6 +316,114 @@ const UnknownRecord = React.lazy(() =>
+      exact
+      component={EditCustomRevenue}
+    />
+    {/* Redirect old routes to nested finances routes for backwards compatibility */}
```
This makes sense but are there specific links we are trying to preserve? Our previous strategy has been to just delete them. Or would it make sense to make a separate file like "reitiredRoutes" if we keep?

antsgar replied:
My thinking was just ensuring that if someone has a page open it would still work for them, but I liked splitting these into a different file so we can eventually move away from them. If we do remove them I'd also need to check the rest of the files to ensure all the links work properly, which you're right we'll have to do eventually to clean this up

Duncan-Brain replied:
True, I am not too worried about a person on that page. But I understand it is a lot of work to fix all the broken links...

Duncan-Brain commented on packages/webapp/src/Routes.jsx:
```diff
@@ -316,6 +316,114 @@ const UnknownRecord = React.lazy(() =>
 
+const FinancesRoutes = () => (
+  <Switch>
+    <Route path="/finances/transactions" exact component={Finances} />
```
I see that it is scoped in figma this way -- I am just surprised we are getting rid of the concept of a "Finances home" / "Dashboard" with a route of just /finances

Duncan-Brain replied:
True I don't know the right way to do it, since I can potentially see more data visualizations in the future being something Kevin + David might think about. I still personally like keeping the `/finances` home url especially since the title is "finances overview" on the page itself not just transactions. But easily reversed in the future too so I am good with it.

Duncan-Brain commented on packages/webapp/src/Routes.jsx:
```diff
@@ -316,6 +316,114 @@ const UnknownRecord = React.lazy(() =>
+      component={UpdateEstimatedCropRevenue}
+    />
+    <Route path="/finances/revenue_types" exact component={RevenueTypes} />
+    <Route path="/finances/add_sale" exact component={AddSale} />
```
I think we didn't get around to changing these url names during the last update. But wondering if we should be moving all references to "revenue" from "sale" here since we are improving our routes structure.

antsgar replied:
For sure, can do!

antsgar replied:
I only changed this one which is customer facing, because changing the component names and params inside the components will make this PR quite a lot bigger and I want to make it easier to review. I'll open up a follow up PR to clean them up

Duncan-Brain replied:
Sounds good!

Duncan-Brain commented on packages/webapp/src/assets/mixin.scss:
```diff
@@ -21,3 +21,15 @@
   box-shadow: 0px 2px 4px rgba(102, 115, 138, 0.3);
 }
+
+@mixin mobile-breakpoint {
```
This is a great mixin -- super useful. Just wondering if there was further discussion with Loic on the breakpoints we want to use?  I see 415, and 1024 on figma. But I might prefer MUI's defaults https://mui.com/material-ui/customization/breakpoints/ which are closer to this number you have here. (But I overrided these for my PR).

antsgar replied:
Ohhh yes, I should have added a comment in here! I talked about this with Loïc in my 1:1 today and the 415px breakpoint caused the Finances screen to overflow when it was between 415 and 600px, so he agreed to make that breakpoint a little wider. I see that 600px actually matches the MUI breakpoints so we could use those! I could rename this mixins to be something like `sm-breakpoint`

Duncan-Brain replied:
its perfect!

Duncan-Brain commented:
Everything is looking great to me! I left a few inline comments/questions.

The only other outstanding questions/comments were:
1. Side menu was not hiding when on mobile -**FIXED with recent commit**
2. Drawer dropdown items I think are supposed to be white and no bottom borders:
![Screenshot 2023-12-20 at 11 17 33 AM](https://github.com/LiteFarmOrg/LiteFarm/assets/30075973/ef3d8ab0-f8e2-4b4b-af31-36a671012506)
![Screenshot 2023-12-20 at 11 19 57 AM](https://github.com/LiteFarmOrg/LiteFarm/assets/30075973/ff81dbb5-2a0c-487c-a8a7-0b490d795b62)
3. Wondering if the collapsible menu Text should have a link directly to the dashboard. We are kind of making the user have two clicks to get to finances.

Duncan-Brain commented on packages/webapp/src/components/Navigation/NavBar/styles.module.scss:
```diff
@@ -37,6 +39,29 @@
+    margin-bottom: -16px;
+  }
+
+  @include desktop-breakpoint {
```
No action needed but I was curious -- For specifically the display:none css (not other styling) I wondered what you think about the MUI useTheme() hook that was already in Navbar.  It seemed useful to conditionally render based on breakpoints. I expanded its use from what it was before and it seems like a handy and explicit way to show what is going on in the JS versus CSS to make things mobile/responsive.

antsgar replied:
I really like the idea of going the JSX route to hide/show stuff, I agree it seems a lot easier to read. I'll refactor!

antsgar commented:
@Duncan-Brain I think I addressed all your comments, except for the borders in the drawer which I don't see, are you still seeing that one? Might have fixed it at some point?

Duncan-Brain commented on packages/webapp/src/App.jsx:
```diff
@@ -16,20 +16,27 @@
-      </Suspense>
+      {isDesktop && (
+        <Suspense fallback={null}>
+          <SlideMenu history={history} classes={{ container: styles.slideMenu }} />
```
It would be cool if we could move this into the Navigation component. Then the extra suspense wouldn't be needed and it would all be in one place.

antsgar replied:
You're not wrong since this is actual navigation, but I'm hesitant to do this much refactoring at this point to make re-reviews a bit easier, so I'll leave it as is for now!

Duncan-Brain replied:
Sounds good 👍

Duncan-Brain commented on packages/webapp/src/containers/Navigation/SlideMenu.jsx:
```diff
@@ -15,7 +15,7 @@
 import { useSelector } from 'react-redux';
 import PropTypes from 'prop-types';
-import PureSlideMenu from '../../components/Navigation/slideMenu';
+import PureSlideMenu from '../../components/Navigation/SlideMenu';
```
For some reason the folder name is actually lowercase for the component

antsgar replied:
Changed the name to side menu, hopefully that'll do the trick

Duncan-Brain commented on packages/webapp/src/stories/Navigation/NavBarDrawer.stories.jsx:
```diff
@@ -1,7 +1,7 @@
 import TopMenu from '../../components/Navigation/Menus/TopMenu';
 import { componentDecoratorsWithoutPadding } from '../Pages/config/Decorators';
-import SlideMenu from '../../components/Navigation/slideMenu';
+import SlideMenu from '../../components/Navigation/SlideMenu';
```
same here with the lowercase naming

Duncan-Brain commented:
Thanks @antsgar for taking a look at my comments! 

I added a few new inline comments since the merge commit. Only the imports might cause problems.

I am still seeing some small styling things with the drawer menu item borders and the selected item behaviour seems to change with a few clicks (see below):
![Screenshot 2023-12-20 at 4 50 17 PM](https://github.com/LiteFarmOrg/LiteFarm/assets/30075973/3957b380-f72a-4851-af9d-327bacbe6d8e)
![Screenshot 2023-12-20 at 4 50 33 PM](https://github.com/LiteFarmOrg/LiteFarm/assets/30075973/c410a0a6-024f-4db6-9517-c9a7bc66d14a)

I noticed a bug in my TopMenu component creating a console error when I useState(null) which should be useState(false). Wondering if I should create a new ticket for the small change or if you can toss it in here too 😂  lol sorry.

Duncan-Brain commented on packages/webapp/src/components/Navigation/utils.jsx:
```diff
@@ -0,0 +1,61 @@
+
+export const getMenuList = (isAdmin) => {
+  const list = [
+    { label: i18n.t('MENU.MAP'), icon: <MapIcon />, path: '/map' },
```
Wondering if this is the cause of the menu text problems. Normally we use the useTranslation hook. Maybe t needs to be passed in from the component.

SayakaOno reviewed (commented):
The menu looks great!

Here are things that I found (an issue in beta included!):
- filters (task, crops)
  There is a line below the toolbar that is partially hidden.
![Screenshot 2023-12-20 at 1 48 15 PM](https://github.com/LiteFarmOrg/LiteFarm/assets/33141219/61708b1a-1c4a-4802-b2ec-43d89efcbaec)
- styling
<img width="764" alt="Screenshot 2023-12-20 at 1 56 20 PM" src="https://github.com/LiteFarmOrg/LiteFarm/assets/33141219/7efbec2f-e3d9-4f3f-bcb0-b7f604a89a6a">

   - the font should be bolder?
   - "farm settings" should be in one line?
   - the menu should be scrollable? I think some items will be hidden on my mac when we add livestock
 - map's drawer menu is broken
![Screenshot 2023-12-20 at 2 01 25 PM](https://github.com/LiteFarmOrg/LiteFarm/assets/33141219/6d32d14c-2536-4045-aa28-bc242f7fccd8)
- Change to `slideMenu` -> `SlideMenu`
   This change doesn't seem to be tracked by Git... it seems working fine, but I'm worried since I have had some problems with capitalization in the past.

antsgar commented:
@Duncan-Brain re

> I am still seeing some small styling things with the drawer menu item borders and the selected item behaviour seems to change with a few clicks (see below):

I'm still not seeing the borders and I double checked and the items have a border: none set 🙈 @SayakaOno are you seeing them?

Re the selected item behavior changing, could you do a video or explain that one a little bit more? 🙏

antsgar commented:
@SayakaOno addressed your comments! Thank you for finding the bug with the map drawer. Synced with Loïc about it and we landed on using the drawer/modal pattern we use for Finances. Can you re-review?

Duncan-Brain commented on packages/webapp/src/App.jsx:
```diff
@@ -16,20 +16,27 @@
 import { SnackbarProvider } from 'notistack';
 import { NotistackSnackbar } from './containers/Snackbar/NotistackSnackbar';
 import { OfflineDetector } from './containers/hooks/useOfflineDetector/OfflineDetector';
 import SlideMenu from './containers/Navigation/SlideMenu';
```
```suggestion
import SlideMenu from './containers/Navigation/SideMenu';
```

Duncan-Brain commented:
@antsgar sorry I was using imprecise language there is no border lol. I didnt process that you couldn't see it so I described just how it looks. I am using the pixel 5 option on chrome tools. And it is the subList CSS that is causing what I am seeing

In the last imagea the top big green space is the padding and the 1 px line must just be a quirk of spacing -- subList needs to have a white background-color. Will make the suggested change.

Heres the video of the behaviour, seems to work as expected on first time and refresh.

https://github.com/LiteFarmOrg/LiteFarm/assets/30075973/598b18c6-4c98-4fb3-9ce8-207569aad637

Duncan-Brain commented on packages/webapp/src/components/Navigation/SideMenu/styles.module.scss:
```diff
@@ -0,0 +1,142 @@
+
+.subList {
+  padding: 8px 0;
+}
```
```suggestion
background-color: var(--bgInputListTile)
}
```

SayakaOno commented:
@antsgar map's drawer looks great!
![Screenshot 2023-12-21 at 8 20 26 AM](https://github.com/LiteFarmOrg/LiteFarm/assets/33141219/756e9fbc-f8b2-4523-ad1e-d898182e0b1d)
I think some styling adjustment is needed, the cursor should be pointer, and the plus button should be bigger.

- this is super small, but the crops icon's path `fill` colour also needs to be changed when the icon colour is not the original one.
![Screenshot 2023-12-21 at 8 33 04 AM](https://github.com/LiteFarmOrg/LiteFarm/assets/33141219/5b2e4c2e-949c-479e-9452-6cbab06524c2)

antsgar commented:
@Duncan-Brain ohhh what you're seeing is the hover effect which for some reason might be behaving weirdly in mobile view? Mobile doesn't need hover anyway so I'll just remove it

antsgar commented:
@SayakaOno @Duncan-Brain I think it's ready for another re-review when you can 🙂

Duncan-Brain reviewed (approved):
Looks good to me! 

Chevron Arrow is missing on Finances drawer when not clicked but I have approved so not a blocker for me:
![Screenshot 2023-12-21 at 1 54 25 PM](https://github.com/LiteFarmOrg/LiteFarm/assets/30075973/57dd8681-42f3-4b86-95fa-df0e4c02a224)

SayakaOno commented on packages/webapp/src/components/MapDrawer/styles.module.scss:
I saw you removed some single quotes, but the rest should be removed as well?
It would be great if you could add license!

SayakaOno reviewed (approved):
My comment can be addressed in the next PR, please merge if that works better for you!

antsgar commented:
@SayakaOno @Duncan-Brain cool thanks, I'll address both those fixes in the next PR! 🙂
